src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryParser.cs

```diff
@@ -0,0 +1,1048 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace System.Runtime.Serialization.Formatters.Binary
+{
+    internal sealed class BinaryParser
+    {
+        private const int ChunkSize = 4096;
+        private static readonly Encoding s_encoding = new UTF8Encoding(false, true);
+
+        internal ObjectReader _objectReader;
+        internal Stream _input;
+        internal long _topId;
+        internal long _headerId;
+        internal SizedArray _objectMapIdTable;
+        internal SizedArray _assemIdToAssemblyTable;    // Used to hold assembly information        
+        internal SerStack _stack = new SerStack("ObjectProgressStack");
+
+        internal BinaryTypeEnum _expectedType = BinaryTypeEnum.ObjectUrt;
+        internal object _expectedTypeInformation;
+        internal ParseRecord _PRS;
+
+        private BinaryAssemblyInfo _systemAssemblyInfo;
+        private BinaryReader _dataReader;
+        private SerStack _opPool;
+
+        private BinaryObject _binaryObject;
+        private BinaryObjectWithMap _bowm;
+        private BinaryObjectWithMapTyped _bowmt;
+
+        internal BinaryObjectString objectString;
```

Comment:
>objectString [](start = 36, length = 12)

 more _'s ? more importantly -- is it worth making some of these internal fields into properties - some could be getters-only perhaps. no need to do it right now, though